src/test/kotlin/org/rust/ide/intentions/SpecifyTypeExplicitlyIntentionTest.kt

```diff
@@ -257,4 +257,33 @@ class SpecifyTypeExplicitlyIntentionTest : RsIntentionTestBase(SpecifyTypeExplic
         """ fn main() { let ref mut var/*caret*/ = 42; } """,
         """ fn main() { let ref mut var: i32 = 42; } """
     )
+
+    fun `test import shadowed type`() = doAvailableTest("""
+        pub mod m1 {
+            pub struct S;
+            pub fn foo() -> S { S }
+        }
+        pub mod m2 { pub struct S; }
+        use m2::S;
+        use crate::m1::foo;
+        fn main() { let s/*caret*/ = foo(); }
+    """, """
+        pub mod m1 {
+            pub struct S;
+            pub fn foo() -> S { S }
+        }
+        pub mod m2 { pub struct S; }
+        use m2::S;
+        use crate::m1::foo;
+        fn main() { let s: crate::m1::S = foo(); }
+    """, preview = """
```

Comment:
The `preview` parameter is for cases when preview differs from actual intention result, in this case preview and result equals and thus `preview` should be omitted

Reply:
Fixed.